guess.html

```diff
@@ -1,38 +1,42 @@
 <script>
-var userNumber; /*Storing user's number*/
-var	userReply;/*Storing Y or N for binumeral selection*/
-var currentPlace=4.5;/*initial is 4.5, but will be average of range*/
-var range_min=0; /*current possible minimum*/
-var range_max=10; /*current possible maximum*/
-var magicnumber=2;/* used to divide and narrow the range*/
+
+var userNumber, /*Storing user's number*/
+	userReply,/*Storing Y or N for binumeral selection*/
+	currentPlace=4.5,/*initial is 4.5, but will be average of range*/
+	range_min=0, /*current possible minimum*/
+	range_max=10; /*current possible maximum*/
 
 /* this function is called to narrow down the possibile ranges.  It uses the userReply answer to narrow down the choices*/
-function narrow()	{ 
-	if (userReply == "Y"){
+function narrow() { 
+	var magicnumber=2;/* used to divide and narrow the range*/
+	if (userReply == "Y") {
 		range_min=currentPlace; currentPlace = (range_max + range_min) / magicnumber;
 		}
-	else {
+	else if (userReply == "N") {
 		range_max=currentPlace; currentPlace = (range_min + range_max) / magicnumber;
 		}
+	else {
+		alert ("You needed to follow the rules! Type Y or N please.")
+	}	
 	};
 
 /* this function is called to count how many possible numbers fall within the set range.*/
-function legalNumbers()	{
+function legalNumbers() {
 	var counter=0; /*counter for numbers that fall in range*/
 	var checkednumber = range_min; /*checking number vs range*/
-	while (checkednumber < range_max){
-		counter++; checkednumber++;}
+	while (checkednumber < range_max) {
+		counter++; checkednumber++;
+	}
 	return counter;
 };
 
 userNumber= prompt("Please enter a whole number between 0 and 10 and I will try to guess it.  Don't worry, I won't really peek, which means I won't cheat so please don't cheat: ");
 
-
 var i=1;
-while (i!=-1){ /*breaking out of this loop if legalNumbers returns 2*/
-userReply = prompt("Is the number bigger than " + currentPlace + " (Y/N)");
-narrow("userReply");
-	if (legalNumbers ( range_min, range_max )==2){
+while (i!=-1) { /*breaking out of this loop if legalNumbers returns 2*/
+userReply = prompt("Is the number bigger than " + currentPlace + " (Y/N)").toUpperCase();
+narrow(userReply);
+	if (legalNumbers (range_min, range_max)==2) {
```

Comment:
instead of using i, you could assign legalNumbers (range_min, range_max) to a variable and stop when the value is two, like this:
```
var legalNumbersResult;
while (legalNumbersResult !== 2) {
  userReply = prompt("Is the number bigger than " + currentPlace + " (Y/N)").toUpperCase();
  narrow(userReply);
  legalNumbersResult = legalNumbers(range_min, range_max);
};
```